mountpoint-s3/src/prefetch.rs

```diff
@@ -398,13 +398,13 @@ impl<E: std::error::Error + Send + Sync> RequestTask<E> {
 #[derive(Debug, Error)]
 pub enum PrefetchReadError<E: std::error::Error> {
     #[error("get request failed")]
-    GetRequestFailed(#[from] E),
+    GetRequestFailed(#[source] E),
```

Comment:
`#[from] E` was a blanket conversion from _any_ type that implements `Error`, so conflicts with any other variant with a `#[from]` from a type that implements `Error`. If we'd just left this as `#[from] E`, and we called `.into()` on an `IntegrityError`, it would be ambiguous to convert to `PrefetchReadError::Integrity` or `PrefetchReadError::GetRequestFailed` (because `IntegrityError` implements `Error`).